pkg/environment/config.go

```diff
@@ -36,12 +45,17 @@ func GetConfig() (base Config, err error) {
 	if err != nil {
 		return
 	}
-	base, err = getConfig(env.RunningMode, baseExtendedURL)
+
+	opts := httpClientOption{
```

Comment:
I was actually trying to push you to pass the Client itself  not config.

Reply:
We don't need an option struct. Just create the client at the top level then pass it around. Imagine now if u added a new flag to the options. Then u will have to go everywhere and modify the client creation to actually use this flag. Right?

But if you create the Client onetime at the top level, then pass it around and later decided to change one of the client options, u can do it only in this one place